stumpy/mstump.py

```diff
@@ -59,7 +59,7 @@ def _multi_mass(Q, T, m, M_T, Σ_T):
     return D
 
 
-def _get_first_mstump_profile(start, T, m, excl_zone, M_T, Σ_T):
+def _get_first_mstump_profile(start, T_A, T_B, m, excl_zone, M_T, Σ_T):
```

Comment:
This implies that we are prepared to perform AB-joins in addition to self-joins?

Reply:
For the first mstump profile yes, then we can directly use this function.

Currently I implemented it here to implement `nan`/`inf` handling the same way as in `stump`. There (for the _first_ subsequence) all `nans` in `T_A` are replaced with zeros, while `T_B` still contains them, because otherwise there is no way to pass this information to `core.mass`.

To actually support AB-joins, there also have to be done some changes in `_stump` itself that are currently a bit out of range.

Reply:
Frankly, I'm not sure what an AB-join for multi-dimensional matrix profile actually means. I don't remember the original authors discussing it and I am concerned with supporting this for users if we haven't thought through it.

> To actually support AB-joins, there also have to be done some changes in _stump itself that are currently a bit out of range.

I'm not sure I understand, technically, `_stump` already supports AB-joins. Or, do you mean AB-joins that contain `nan`/`inf` in both sequences?

Reply:
Alright, I think I am okay with this since it is a private function